add alternative vsix archive publishing and separate build and publish workflows

ndoschek opened the pull request:
add alternative vsix archive publishing 

- Add compress-vsix.js script to create a tar.gz archive of all .vsix files
- Update get-external-builtins.js to also download pre-built .vsix from open-vsx.org if available
- Update vsix packaging to use consistent publisher.name.vsix filename format
- Remove theia-testing.js script in favor of archive-based testing workflow
- Update documentation for new build, publish, and testing workflows
- Fix missing await in archive-source.js

ci: separate build and publish workflows 

- Split build.yml into reusable build workflow and new publish.yml for publishing
- Pin GitHub Actions to specific commit SHAs for security
- Add workflow dispatch option to publish to either open-vsx or vsix archive

Contributed on behalf of STMicroelectronics

ndoschek committed:
add alternative vsix archive publishing

- Add compress-vsix.js script to create a tar.gz archive of all .vsix files
- Update get-external-builtins.js to also download pre-built .vsix from open-vsx.org if available
- Update vsix packaging to use consistent publisher.name.vsix filename format
- Remove theia-testing.js script in favor of archive-based testing workflow
- Update documentation for new build, publish, and testing workflows
- Fix missing await in archive-source.js

Contributed on behalf of STMicroelectronics

ndoschek committed:
ci: separate build and publish workflows

- Split build.yml into reusable build workflow and new publish.yml for publishing
- Pin GitHub Actions to specific commit SHAs for security
- Add workflow dispatch option to publish to either open-vsx or vsix archive

Contributed on behalf of STMicroelectronics

sgraband commented on src/get-external-builtins.js:
```diff
@@ -18,12 +18,18 @@
 const colors = require('colors');
 colors.enable();
+
+const OPEN_VSX_ORG_URL = 'https://open-vsx.org';
```
We could retrieve that from the versions file to avoid duplication

sgraband reviewed (commented):
LGTM, i just have a minor inline suggestion

ndoschek committed:
reuse OPEN_VSX_ORG_URL from version.js

- Export OPEN_VSX_ORG_URL from version.js instead of duplicating it
- Import the constant in get-external-builtins.js

ndoschek committed:
ci: create GitHub release in publish-archive workflow

- Extract VS Code version from vscode/package.json
- Create a new GitHub release tagged with the version
- Attach the vsix archive tar.gz to the release

ndoschek commented:
Thank you for the review @sgraband, I addressed your comment and i added another commit to also already create the GH release via the workflow that no manual steps are needed anymore. (See existing GH release to compare: https://github.com/eclipse-theia/vscode-builtin-extensions/releases/tag/1.104.0)
Could you have another look? TIA

ndoschek committed:
ci: add release notes with IP checks placeholder and contents log

- Capture compress-vsix output to include in release notes
- Add IP checks placeholder comment for manual editing
- Include compressed archive contents in a collapsible details section

sgraband reviewed (approved):
Thank you! LGTM :+1:
